lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -222,7 +221,11 @@ def evaluate_selections(path, scoped_context, owner_object, owner_type, selectio
               end
               after_lazy(app_result, owner: owner_type, field: field_defn, path: next_path, scoped_context: context.scoped_context, owner_object: object, arguments: kwarg_arguments) do |inner_result|
                 continue_value = continue_value(next_path, inner_result, field_defn, return_type.non_null?, ast_node)
-                if HALT != continue_value
+                if RawValue === continue_value
+                  # Write raw value directly to the response without resolving nested objects
+                  write_in_response(next_path, continue_value.resolve)
```

Comment:
Could this use the other method instead? 

```suggestion
                  write_in_response(next_path, continue_value.object)
```